ui/src/app/machines/views/MachineDetails/MachinePCIDevices/MachinePCIDevices.tsx

```diff
@@ -1,29 +1,98 @@
 import { useEffect } from "react";
 
 import { Button, Col, Icon, Row, Strip } from "@canonical/react-components";
+import pluralize from "pluralize";
 import { useDispatch, useSelector } from "react-redux";
 import { useParams } from "react-router";
+import { Link } from "react-router-dom";
 
 import type { SetSelectedAction } from "../MachineSummary";
 
+import DoubleRow from "app/base/components/DoubleRow";
 import Placeholder from "app/base/components/Placeholder";
+import { HardwareType } from "app/base/enum";
 import { useWindowTitle } from "app/base/hooks";
 import type { RouteParams } from "app/base/types";
 import machineSelectors from "app/store/machine/selectors";
+import type { Machine } from "app/store/machine/types";
 import { actions as nodeDeviceActions } from "app/store/nodedevice";
 import nodeDeviceSelectors from "app/store/nodedevice/selectors";
+import { NodeDeviceBus } from "app/store/nodedevice/types";
+import type { NodeDevice } from "app/store/nodedevice/types";
 import type { RootState } from "app/store/root/types";
 import { NodeActions } from "app/store/types/node";
 
+type NodeDeviceGroup = {
+  hardwareTypes: HardwareType[];
+  items: NodeDevice[];
+  label: string;
+  pathname?: string;
+};
+
 type Props = { setSelectedAction: SetSelectedAction };
 
+const generateGroup = (group: NodeDeviceGroup, machine: Machine) =>
+  group.items.map((nodeDevice, i) => {
+    const {
+      commissioning_driver,
+      id,
+      numa_node_id,
+      pci_address,
+      product_id,
+      product_name,
+      vendor_id,
+      vendor_name,
+    } = nodeDevice;
+    const numaNode =
+      "numa_nodes" in machine
+        ? machine.numa_nodes.find((numa) => numa.id === numa_node_id)
+        : null;
+
+    return (
+      <tr key={`pci-device-${id}`}>
```

Comment:
Is this a regular table because it doesn't need any of the MainTable features?

Reply:
Yeah pretty much. I figured if it doesn't need to be sortable and doesn't expand it's probably simpler to just leave it as a normal table.